cli/start.go

```diff
@@ -440,10 +445,17 @@ func runStart(_ *cobra.Command, args []string) error {
 	}()
 
 	select {
-	case <-signalCh:
-		log.Errorf(context.TODO(), "second signal received, initiating hard shutdown")
+	case sig := <-signalCh:
+		err := fmt.Errorf("received signal '%s' during shutdown, initiating hard shutdown", sig)
+		log.Errorf(context.TODO(), "%v", err)
+		// On Unix, os.Signal is syscall.Signal and it's convertible to int.
+		ErrorCode = 128 + int(sig.(syscall.Signal))
+		return err
 	case <-time.After(time.Minute):
-		log.Errorf(context.TODO(), "time limit reached, initiating hard shutdown")
+		err := errors.New("time limit reached, initiating hard shutdown")
+		log.Errorf(context.TODO(), "%v", err)
+		ErrorCode = 2
```

Comment:
I don't think that a timeout during an attempted shutdown changes the fact that we are exiting due to an earlier signal (nor, for that matter, does completing the graceful shutdown process). I think that we should save the code of the first signal, and exit with that status unless we get a second signal. I'm not sure it ever makes sense for a server like ours to exit with a zero status code (maybe for `cockroach quit`, but I probably wouldn't do it even then).